ark/phenotyping/som_runner.R

```diff
@@ -0,0 +1,36 @@
+library(data.table)
+library(FlowSOM)
+
+print("Reading command args")
+args <- commandArgs(trailingOnly=TRUE)
+
+# get path to pixel mat data
+print("Getting pixel mat data path")
+pixelMatPath <- args[1]
+
+# create a vector out of the list of channels provided
+print("Creating marker list")
+markers <- unlist(strsplit(args[2], split=","))
+print(markers)
+
+# get the path to the write directory
+print("Getting the write directory")
+write_dir <- args[3]
+
+# read the pixel mat data
+print("Reading the pixel matrix data")
+pixelMatData <- fread(pixelMatPath, select=markers)
+
+# subset the data for SOM clustering only on the channels we're interested in
+
+# run the SOM clustering step
+print("Run the SOM clustering")
+som_results <- SOM(data=as.matrix(pixelMatData))
```

Comment:
This looks like the top-level SOM function. This trains a network and evaluates it on the training data. 

However, we want to separate this into a training the network step and an evaluating the network on data step. Otherwise, we won't ever be able to subset the data, train on the subset and evaluate on the whole dataset.